proto-public/pbmesh/v1alpha1/pbproxystate/traffic_permissions.proto

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1.pbproxystate;
+
+message L7TrafficPermission {}
+
+enum TrafficPermissionAction {
+  INTENTION_ACTION_DENY = 0;
+  INTENTION_ACTION_ALLOW = 1;
+}
+
+message L4TrafficPermission {
+  TrafficPermissionAction default_action = 1;
+  // permission is constructed by pre-flattening all relevant permissions.
+  L4Permission permission = 2;
+}
+
+message L4Permission {
+  repeated L4Source sources = 1;
+  repeated L4Source exclude_sources = 2;
+
+  // We don't need destination rules here because they either apply to L7 features or multi-ports.
+  // In the case of multi-ports, the proxystateconverter is responsible for filtering per-port intentions.
+}
+
+message L4Source {
+  repeated string spiffe_ids = 1;
```

Comment:
nit: technically these don't have to be valid spiffe IDs as they can be regexes for wildcard sources. I'd probably lean towards calling them `principals` same as envoy.

Reply:
For now, I updated this to be spiffe regex since regular expressions are a specific type of principal. I also updated the name "source" to "principal" which corrolates one to one with the Envoy principals that we generate.